### PR TITLE
[FLINK-34533][release] Add release note for version 1.19

### PR DESCRIPTION
**Pull request by lincoln-lil**
## What is the purpose of the change

This pull request adds release note for version 1.19.

## Brief change log

Add release note for 1.19.

## Verifying this change

This change is a doc-only without test coverage.

## Does this pull request potentially affect one of the following parts:

  - Dependencies (does it add or upgrade a dependency): (yes / no)
  - The public API, i.e., is any changed class annotated with `@Public(Evolving)`: (yes / no)
  - The serializers: (yes / no / don't know)
  - The runtime per-record code paths (performance sensitive): (yes / no / don't know)
  - Anything that affects deployment or recovery: JobManager (and its components), Checkpointing, Kubernetes/Yarn, ZooKeeper: (yes / no / don't know)
  - The S3 file system connector: (yes / no / don't know)

## Documentation

  - Does this pull request introduce a new feature? (yes / no)
  - If yes, how is the feature documented? (not applicable / docs / JavaDocs / not documented)

### CODE REVIEW

**Comment by yuchen-ecnu** [comment]
Hi @lincoln-lil, so sorry for the delay in resolving the last [ticket](https://issues.apache.org/jira/browse/FLINK-33436) of FLIP-375 until recently. Probably because I didn't fill in the release note in the ticket of FLIP-375, so this part is not in the PR, can you help me to add this part?

Here is the content of the release note for reference.

```markdown
#### Supports profiling JobManager/TaskManager with Async-profiler on Flink Web

##### [FLINK-33325](https://issues.apache.org/jira/browse/FLINK-33325)

In Flink 1.19, we support trigger the JobManager/TaskManager level profiling, which allows users to create a profiling instance with arbitrary intervals and events mode (supported by [async-profiler](https://github.com/async-profiler/async-profiler)).
Flink users can complete the profiling submission and result export in the Flink Web UI conveniently.

For example, 
- First, users should find out the candidate TaskManager/JobManager with performance bottleneck for profiling, and switch to the corresponding TaskManager/JobManager page (profiler tab).
- Users can submit a profiling instance with a specified period and mode by simply clicking on the button `Create Profiling Instance`. (The description of the profiling mode will be shown when hovering over the corresponding mode.)
- Once the profiling instance is complete, the user can easily download the interactive HTML file by clicking on the link.

**More Information**
- [Documents](https://nightlies.apache.org/flink/flink-docs-release-1.19/docs/ops/debugging/profiler/)
- [FLIP-375: Built-in cross-platform powerful java profiler](https://cwiki.apache.org/confluence/x/64lEE)
```
flip375_example

**Comment by lincoln-lil** [comment]
@yuchen-ecnu Thanks for your updates! Don't worry, we haven't release 1.19 yet.
I've added the description(the image is useful for the release announcement but usually not included in the release note).

File: docs/content.zh/release-notes/flink-1.19.md [hunk 1]
```diff
@@ -0,0 +1,354 @@
...
+
+In Flink 1.19, we have supported dynamic source parallelism inference for batch jobs, which allows
+source connectors to dynamically infer the parallelism based on the actual amount of data to consume.
+This feature is a significant improvement over previous versions, which only assigned a fixed default
+parallelism to source vertices.
+Source connectors need to implement the inference interface to enable dynamic parallelism inference.
+Currently, the FileSource connector has already been developed with this functionality in place.
+Additionally, the configuration `execution.batch.adaptive.auto-parallelism.default-source-parallelism`
+will be used as the upper bound of source parallelism inference. And now it will not default to 1.
+Instead, if it is not set, the upper bound of allowed parallelism set via
```
Comment (JingGe) [inline docs/content.zh/release-notes/flink-1.19.md:155]:
```suggestion
If it is not set, the upper bound of allowed parallelism set via
```

File: docs/content.zh/release-notes/flink-1.19.md [hunk 2]
```diff
@@ -0,0 +1,354 @@
...
+`execution.batch.adaptive.auto-parallelism.max-parallelism` will be used. If that configuration is
```
Comment (JingGe) [inline docs/content.zh/release-notes/flink-1.19.md:156]:
```suggestion
`execution.batch.adaptive.auto-parallelism.max-parallelism` will be used instead. If that configuration is
```

**Review by JingGe — COMMENTED** [review]
Thanks @lincoln-lil for driving it! Just left some comments, PTAL

**Review by lincoln-lil — COMMENTED** [review]
@JingGe Thank you very much for detailed review! I've updated pr according to your comments.

File: docs/content.zh/release-notes/flink-1.19.md [hunk 3]
```diff
@@ -0,0 +1,354 @@
...
+`pipeline.serialization-config`, `pipeline.force-avro`, `pipeline.force-kryo`, and `pipeline.generic-types`.
+Registration of instance-level serializers is deprecated, using class-level serializers instead.
+
+#### Migrate string configuration key to ConfigOption
+
+##### [FLINK-34079](https://issues.apache.org/jira/browse/FLINK-34079)
+
+We have deprecated all setXxx and getXxx methods except `getString(String key, String defaultValue)`
+and `setString(String key, String value)`, such as: `setInteger`, `setLong`, `getInteger` and `getLong` etc.
+We strongly recommend users and developers use get and set methods directly.
```
[the remark this answers is not in the document]
Comment (1996fanrui) [reply docs/content.zh/release-notes/flink-1.19.md]: Thanks for the good suggestion and ping.

Proposed change looks good to me.

File: docs/content.zh/release-notes/flink-1.19.md [hunk 4]
```diff
@@ -0,0 +1,354 @@
...
+In addition, we recommend users to use ConfigOption instead of string as key.
+
+#### Support System out and err to be redirected to LOG or discarded
+
+##### [FLINK-33625](https://issues.apache.org/jira/browse/FLINK-33625)
+
+`System.out` and `System.err` output the content to the `taskmanager.out` and `taskmanager.err` files.
+In a production environment, if flink users use them to print a lot of content, the limits of yarn
+or kubernetes may be exceeded, eventually causing the TaskManager to be killed. Flink supports
+redirecting the `System.out` and `System.err` to the log file, and the log file can be rolled to
```
[the remark this answers is not in the document]
Comment (1996fanrui) [reply docs/content.zh/release-notes/flink-1.19.md:112]: One tm might keep latest 10 log files

File: docs/content.zh/release-notes/flink-1.19.md [hunk 5]
```diff
@@ -0,0 +1,354 @@
...
+
+#### Support dynamic source parallelism inference for batch jobs
+
+##### [FLINK-33768](https://issues.apache.org/jira/browse/FLINK-33768)
+
+In Flink 1.19, we have supported dynamic source parallelism inference for batch jobs, which allows
+source connectors to dynamically infer the parallelism based on the actual amount of data to consume.
+This feature is a significant improvement over previous versions, which only assigned a fixed default
+parallelism to source vertices.
+Source connectors need to implement the inference interface to enable dynamic parallelism inference.
```
Comment (ruanhang1993) [inline docs/content.zh/release-notes/flink-1.19.md:151]: `Source connectors need to implement the interface ...`. There is a useless word 'inference'.

File: docs/content.zh/release-notes/flink-1.19.md [hunk 6]
```diff
@@ -0,0 +1,354 @@
...
+`RestoreMode#LEGACY` is deprecated. Please use `RestoreMode#CLAIM` or `RestoreMode#NO_CLAIM` mode
+instead to get a clear state file ownership when restoring.
+
+#### CheckpointsCleaner clean individual checkpoint states in parallel
+
+##### [FLINK-33090](https://issues.apache.org/jira/browse/FLINK-33090)
+
+Now when disposing of no longer needed checkpoints, every state handle/state file will be disposed
+in parallel by the ioExecutor, vastly improving the disposing speed of a single checkpoint (for
+large checkpoints, the disposal time can be improved from 10 minutes to < 1 minute) . The old
```
Comment (snuyanzin) [inline docs/content.zh/release-notes/flink-1.19.md:54]:
```suggestion
large checkpoints, the disposal time can be improved from 10 minutes to < 1 minute). The old
```

File: docs/content.zh/release-notes/flink-1.19.md [hunk 7]
```diff
@@ -0,0 +1,354 @@
...
+
+#### Deprecate RichFunction#open(Configuration parameters)
+
+##### [FLINK-32978](https://issues.apache.org/jira/browse/FLINK-32978)
+
+The `RichFunction#open(Configuration parameters)` method has been deprecated and will be removed in
+future versions. Users are encouraged to migrate to the new `RichFunction#open(OpenContext openContext)`
+method, which provides a more comprehensive context for initialization.
+Here are the key changes and recommendations for migration:
+The `open(Configuration parameters)` method is now marked as deprecated.A new method `open(OpenContext openContext)`
```
Comment (snuyanzin) [inline docs/content.zh/release-notes/flink-1.19.md:239]:
```suggestion
The `open(Configuration parameters)` method is now marked as deprecated. A new method `open(OpenContext openContext)`
```

File: docs/content.zh/release-notes/flink-1.19.md [hunk 8]
```diff
@@ -0,0 +1,354 @@
...
+corresponding value. At the same time, if non-essential parameters are not specified, they will default to being filled with null.
+
+#### Window TVF Aggregation Supports Changelog Inputs
+
+##### [FLINK-20281](https://issues.apache.org/jira/browse/FLINK-20281)
+
+The Window aggregation operator (produced by Window TVF) can consume a changelog stream generated by
+nodes such as a CDC connector.
+
+#### Supports  SESSION Window TVF in Streaming Mode
```
Comment (snuyanzin) [inline docs/content.zh/release-notes/flink-1.19.md:327]:
```suggestion
#### Supports SESSION Window TVF in Streaming Mode
```

File: docs/content.zh/release-notes/flink-1.19.md (hunk 5 above)
Comment (lincoln-lil) [reply docs/content.zh/release-notes/flink-1.19.md:151]: @ruanhang1993 IIUC, the 'inference interface' refers to  `DynamicParallelismInference`(https://cwiki.apache.org/confluence/display/FLINK/FLIP-379%3A+Dynamic+source+parallelism+inference+for+batch+jobs), so it's needed here.

**Comment by lincoln-lil** [comment]
@snuyanzin Thanks for reviewing this! I've fixed these incorrect spaces.

**Comment by snuyanzin** [comment]
@lincoln-lil thanks
 a question: since  tests are now executed against jdk 8, 11, 17, 21
should we add a note about jdk21 Beta support ?

**Comment by lincoln-lil** [comment]
@snuyanzin This should be added, I'll add https://issues.apache.org/jira/browse/FLINK-33163.

File: docs/content.zh/release-notes/flink-1.19.md [hunk 9]
```diff
@@ -0,0 +1,362 @@
...
+
+#### Drop support for python 3.7
+
+##### [FLINK-33029](https://issues.apache.org/jira/browse/FLINK-33029)
+
+#### Add support for python 3.11
+
+##### [FLINK-33030](https://issues.apache.org/jira/browse/FLINK-33030)
+
+## Build System
```
Comment (lincoln-lil) [inline docs/content.zh/release-notes/flink-1.19.md:39]: @snuyanzin I've added the java 21 support, PTAL, thanks : )
(also add it to the release announcement pr:
https://github.com/apache/flink-web/pull/721/files#diff-9a4ae63f4269f0b6dee7a38b30564cc94f59cd21f1e5eb6edb13902c1f8bb3fbR234)

File: docs/content.zh/_index.md [hunk 10]
```diff
@@ -85,7 +85,8 @@ under the License.
 For some reason Hugo will only allow linking to the 
 release notes if there is a leading '/' and file extension.
 -->
-请参阅 [Flink 1.18]({{< ref "/release-notes/flink-1.18.md" >}}),
+请参阅 [Flink 1.19]({{< ref "/release-notes/flink-1.19.md" >}}),
+[Flink 1.18]({{< ref "/release-notes/flink-1.18.md" >}}),
```
Comment (PatrickRen) [inline docs/content.zh/_index.md:89]: nit: should use Chinese comma “，”

File: docs/content.zh/release-notes/flink-1.19.md [hunk 11]
```diff
@@ -0,0 +1,362 @@
...
+
+#### Deprecate API that uses Flink's Time implementation (related to FLINK-14638)
+
+##### [FLINK-32570](https://issues.apache.org/jira/browse/FLINK-32570)
+
+Flink's Time classes are deprecated now and will be subject to deletion with the release of Flink 2.0.
+Please start to use Java's own `Duration` class, instead. Methods supporting the `Duration` class
+that replace the deprecated Time-based methods were introduced.
+
+#### Add new interfaces for SinkV2 to synchronize the API with the SourceV2 API
```
Comment (PatrickRen) [inline docs/content.zh/release-notes/flink-1.19.md:262]: There's no `SourceV2 API` I think. `Source API` should be good

**Review by PatrickRen — COMMENTED** [review]
@lincoln-lil Thanks for the work! I left some comments.

**Comment by lincoln-lil** [comment]
@PatrickRen Thanks for reviewing this! I've addressed your comments and updated the pr.

File: docs/content.zh/release-notes/flink-1.19.md [hunk 12]
```diff
@@ -0,0 +1,362 @@
...
+
+### Table SQL / API
+
+#### Support Setting Parallelism for Table/SQL Sources
+
+##### [FLINK-33261](https://issues.apache.org/jira/browse/FLINK-33261)
+
+Scan table sources can now be set a custom parallelism for performance tuning via the `scan.parallelism`
+option. Currently, only the DataGen connector has been adapted to support that, Kafka connector is
+on the way. Please check [scan-table-source](https://nightlies.apache.org/flink/flink-docs-master/docs/dev/table/sourcessinks/#scan-table-source) on how to adapt your custom connectors to it.
```
Comment (Myasuka) [inline docs/content.zh/release-notes/flink-1.19.md:293]: I think we should align with the announcement blog to change the doc link from `flink-docs-master` to `flink-docs-release-1.19`.

File: docs/content.zh/release-notes/flink-1.19.md [hunk 13]
```diff
@@ -0,0 +1,362 @@
...
+Flink introduces `env.java.opts.sql-gateway` for specifying the Java options for the SQL Gateway,
+which allows you to fine-tune the memory settings, garbage collection behavior, and other relevant Java parameters.
+
+#### Support Configuring Different State TTLs using SQL Hint
+
+##### [FLINK-33397](https://issues.apache.org/jira/browse/FLINK-33397)
+
+This is a new feature in Apache Flink 1.19 that enhances the flexibility and user experience when
+managing SQL state time-to-live (TTL) settings. Users can now specify custom TTL values for regular
+joins and group aggregations directly within their queries by [utilizing the STATE_TTL hint](https://nightlies.apache.org/flink/flink-docs-master/docs/dev/table/sql/queries/hints/#state-ttl-hints).
```
Comment (Myasuka) [inline docs/content.zh/release-notes/flink-1.19.md:308]: Same here, I think using the link of `https://nightlies.apache.org/flink/flink-docs-release-1.19/docs/dev/table/sql/queries/hints/#state-ttl-hints` could be better.

**Review by Myasuka — COMMENTED** [review]
Thanks for creating this PR, PTAL.

**Comment by lincoln-lil** [comment]
@Myasuka I've updated all doc links to use `flink-docs-release-1.19`. Thanks for revewing.

File: docs/content.zh/release-notes/flink-1.19.md (hunk 9 above)
Comment (snuyanzin) [reply docs/content.zh/release-notes/flink-1.19.md:39]: great, thanks for adding it

**Review by ferenc-csaky — APPROVED** [review]

**Comment by lincoln-lil** [comment]
Will merge this.

**Pull request merged**